include/modem/location.h

```diff
@@ -0,0 +1,279 @@
+enum loc_accuracy {
+	/** Allow lower accuracy to conserve power. */
+	LOC_ACCURACY_LOW,
+	/** Normal accuracy. */
+	LOC_ACCURACY_NORMAL,
+	/** Allow higher power consumption to improve accuracy. */
+	LOC_ACCURACY_HIGH
+};
+
+/** Positioning service. */
+enum loc_service {
+	/**
+	 * @brief Use any location service that has been configured to be available.
+	 *
+	 * @details This is especially useful when only one service is configured but can be used
+	 * even if many are available.
+	 */
+	LOC_SERVICE_ANY,
+	/** nRF Cloud location service. */
+	LOC_SERVICE_NRF_CLOUD,
+	/**Here location service. */
+	LOC_SERVICE_HERE,
+	/**Skyhook location service. */
+	LOC_SERVICE_SKYHOOK
+};
+
+/** Date and time (UTC). */
+struct loc_datetime {
+	/** True if date and time are valid, false if not. */
+	bool valid;
+	/** 4-digit representation (Gregorian calendar). */
+	uint16_t year;
+	/** 1...12 */
+	uint8_t month;
+	/** 1...31 */
+	uint8_t day;
+	/** 0...23 */
+	uint8_t hour;
+	/** 0...59 */
+	uint8_t minute;
+	/** 0...59 */
+	uint8_t second;
+	/** 0...999 */
+	uint16_t ms;
+};
+
+/** Location data. */
+struct loc_location {
+	/** Geodetic latitude (deg) in WGS-84. */
+	double latitude;
+	/** Geodetic longitude (deg) in WGS-84. */
+	double longitude;
+	/** Location accuracy in meters (1-sigma). */
+	float accuracy;
+	/** Date and time (UTC). */
+	struct loc_datetime datetime;
+};
+
+/** Error cause. */
+struct loc_cause {
+	/* TODO: Error codes? */
+	uint8_t error;
+};
+
+/** Location event data. */
+struct loc_event_data {
+	/** Event ID. */
+	enum loc_event_id id;
+
+	/** Used positioning method. */
+	enum loc_method method;
+
+	union {
+		/** Current location, used with event LOC_EVT_LOCATION. */
+		struct loc_location location;
+		/** Error cause, used with event LOC_EVT_ERROR. */
+		struct loc_cause cause;
+	};
+};
+
+/** GNSS configuration. */
+struct loc_gnss_config {
+	/**
+	 * @brief Timeout (in seconds), i.e. how long GNSS is allowed to run when trying to
+	 * acquire a fix.
+	 *
+	 * @details Note that this is not real time as experienced by the user.
+	 * Since GNSS cannot run while LTE is operating, the running time is
+	 * counted from the instant when GNSS is allowed to start. If LTE power saving mode (PSM)
+	 * is requested, GNSS waits until the modem enters PSM before starting up, thus maximizing
+	 * uninterrupted operating window and minimizing power consumption.
+	 */
+	uint16_t timeout;
+
+	/** Desired accuracy level. */
+	enum loc_accuracy accuracy;
+
+	/**
+	 * @brief If accuracy is set to LOC_ACCURACY_HIGH, allow GNSS to attempt
+	 * num_consecutive_fixes fixes after the first successful fix before outputting the
+	 * current location.
+	 *
+	 * @details This typically improves the location accuracy. If accuracy is set to
+	 * LOC_ACCURACY_NORMAL or LOC_ACCURACY_LOW this parameter has no effect.
+	 */
+	uint8_t num_consecutive_fixes;
+};
+
+/** LTE cellular positioning configuration. */
+struct loc_cellular_config {
```

Comment:
We may need a parameter indicating whether the location is needed back to the device. In most use cases, just sending the data to the cloud is enough.